### ISSUE TITLE
Right controller not working

### ISSUE THREAD

**Issue by 2jackiclark** [msub2/silk-brush#22]
I went into Silk Brush..seems quite pale? Couldn't get past first bit as right controller wasn't working ( but working fine in other applications including Tiltbrush)

**Comment by msub2** [comment]
What headset/browser were you using? Also, what kind of controller issues were you having (button input, tracking, etc)?

**Comment by 2jackiclark** [comment]
I was using a Rift. Cant remember exact wording but the right controller
was stuck on Point to Interact? So just a few blue dots coming out of the
front and not responding.
Hope that helps

Jacki Clark

**Comment by msub2** [comment]
Ahh okay, it sounds like this is the same issue affecting #18 and #19 then. Depending on how the palette is oriented the raycast doesn't reach it correctly, which I'm currently looking into the cause of. Did you try holding it up and angling it down towards you? I find that seems to be a fairly consistent position where it works.

**Comment by 2jackiclark** [comment]
I tried pointing it in various directions with no success. The colours in
the Tiltbrush opening page were very pale too if that is of any significance
B/w

Jacki Clark

**Comment by msub2** [comment]
Interesting, I haven't heard of anyone not being able to get it working at all. I'll see if I can follow up with someone else that has a Rift. The color issue is also new to me; if you could somehow get a picture/record a short video, that'd be helpful.

**Comment by 2jackiclark** [comment]
I will go back on it later today

**Comment by 2jackiclark** [comment]
I had three more goes. Twice it opened up with a pale Tiltbrush page,two
times the right hand controller stuck on Point to Interact, the left stuck
on New file? option. One attempt I actually got past that but right
controller was pointing towards me on its 'back' as if it was doing
backstroke swimming towards me if that makes sense?

**Comment by 2jackiclark** [comment]
My screen recorder wouldnt work with it sadly and what I was seeing in the
headset was not on my laptop screen.

**Comment by De-Panther** [comment]
Does this demo works properly on your computer?
https://de-panther.github.io/unity-webxr-export/Build/

Also, to show the content of the VR on your screen, you can use Oculus Mirror:
https://developer.oculus.com/documentation/native/pc/dg-compositor-mirror/

**Comment by 2jackiclark** [comment]
Will try tomorrow

Jacki Clark

**Comment by 2jackiclark** [comment]
I can access the demo, I see a desert scene and a table in front of me with
a shiny ball and 3 cubes. Ahead of me is the number 90. I can pick up the
ball and cubes and put them on the floor behind me

**Comment by De-Panther** [comment]
With both right and left controllers?

**Comment by 2jackiclark** [comment]
yes and this time I seemed to be able to put the boxes off the front and
side of the table

**Comment by jones1618** [comment]
In Oculus Quest 2 I can open Silk Brush, draw, adjust brush size and bring up the left-hand menu and cycle through the three pages but _cannot_ activate or touch any of the buttons. So, I can't change tools, erase, save, etc.

Normally, can you just touch buttons by moving the right controller or does it require you to click the right trigger or other button to "click." 

FYI: I can use Gravity Sketch just fine

**Comment by msub2** [comment]
@jones1618 You should just have to pull the trigger while the pointer is active over a menu item and it should activate the button, whatever it may be. I'm currently working on fixing the issue of the pointer not always being active when it should be but you should still be able to hit them regardless.

**Comment by jones1618** [comment]
> @jones1618 You should just have to pull the trigger while the pointer is active over a menu item and it should activate the button, whatever it may be. I'm currently working on fixing the issue of the pointer not always being active when it should be but you should still be able to hit them regardless.

I assume you mean right-trigger? So, point at menu item (touching it in 3-space) and then right-trigger should click it. Right?

**Comment by msub2** [comment]
If you can see the pointer circle active on the menu item with the guidelines from the controller, then yes it should. If you're able to get a video clip of it acting otherwise, that would be helpful for my debugging.

**Comment by 2jackiclark** [comment]
Yes

Jacki Clark

**Comment by jones1618** [comment]
> If you can see the pointer circle active on the menu item with the guidelines from the controller, then yes it should. If you're able to get a video clip of it acting otherwise, that would be helpful for my debugging.

OK, I got it to work by tilting my left hand so that the halo was nearly pointed at my face. In other words, the menus don't actiivate unless I'm looking down from the top of them, practically edge-on. It is an awkward angle and would probably be fatiguing for long sessions. However, most everything works and is quite feature rich and fun. Good job on making all this work.

Some of the advanced menu actions crash the app (like polyhedra) but I'll report issues separately.
